lib/parallel.rb

```diff
@@ -641,9 +641,33 @@ def with_instrumentation(item, index, options)
 
     def instrument_finish(item, index, result, options)
       return unless on_finish = options[:finish]
+      return instrument_finish_in_order(item, index, result, options) if options[:finish_in_order]
       options[:mutex].synchronize { on_finish.call(item, index, result) }
     end
 
+    def instrument_finish_in_order(item, index, result, options)
+      options[:mutex].synchronize do
+        options[:finish_items] ||= []
+        options[:finish_items_done] ||= []
+        options[:finish_index] ||= 0
```

Comment:
can these be local variables ?

Reply:
You can't use local variables, as they need to maintain the state between invocations.
As they are being called via as a class method, it can't use instance variables either.